packages/opds-web-client/src/components/OPDSCatalog.tsx

```diff
@@ -18,28 +16,30 @@ export interface OPDSCatalogProps {
   epubReaderUrlTemplate?: (epubUrl: string) => string;
 }
 
-/** The main application component. */
-export default class OPDSCatalog extends React.Component<OPDSCatalogProps, {}> {
-  store: Redux.Store<State>;
-  context: NavigateContext;
+/**
+ * The main application component.
+ *  - Renders root and passes props along with store to root
+ *  - Creates the redux store using OPDSStore
+ *  - Passes the redux store down the tree in context
+ */
+const OPDSCatalog: React.FunctionComponent<OPDSCatalogProps> = (props) => {
 
-  static contextTypes: React.ValidationMap<NavigateContext> = {
-    router: PropTypes.object.isRequired as React.Validator<RouterType>,
-    pathFor: PropTypes.func.isRequired
-  };
+  return (
+    <OPDSStore initialState={props.initialState} authPlugins={props.authPlugins}>
+      <RootWrapper {...props} />
+    </OPDSStore>
+  );
+};
 
-  constructor(props, context) {
-    super(props);
-    this.store = buildStore(this.props.initialState || undefined, this.props.authPlugins || [BasicAuthPlugin], context.pathFor);
-  }
+/**
+ * Simple wrapper to allow store to become available in context
+ * before we try to access it.
+ */
+const RootWrapper: React.FunctionComponent<OPDSCatalogProps> = (props) => {
+  const {store} = useStore();
+  return (
+    <Root store={store} {...props} />
+  );
+};
```

Comment:
Is there a reason why the `RootWrapper` exists. It seems possible to render `Root` in the `OPDSCatalog` function and call `useStore` in there, right? Just wondering why the need for this extra `RootWrapper` function.

Reply:
You might be right actually. I thought you couldn't access the context unless it was separated by an encapsulated component like this, but maybe that was with the old context API or I'm misremembering. Just tried it in a [codesandbox](https://codesandbox.io/s/adoring-pasteur-hu5m5?fontsize=14&hidenavigation=1&theme=dark) without that and it seems to work. I'll remove this